api/rebac/v1/check.proto

```diff
@@ -0,0 +1,31 @@
+syntax = "proto3";
+
+package api.rebac.v1;
+import "google/api/annotations.proto";
+option go_package = "ciam-rebac/api/rebac/v1;v1";
+option java_multiple_files = true;
+option java_package = "api.rebac.v1";
+
+service Check {
+	rpc CheckPermission (CheckPermissionRequest) returns (CheckPermissionResponse) {
+		option (google.api.http) = {
+			post: "/api/authz/v1/check"
+			body: "*"
+		};
+	};
+}
+
+message CheckPermissionRequest {
```

Comment:
Should we roll up these attributes into SubjectReference/ObjectReference as in the following?
https://buf.build/authzed/api/docs/main:authzed.api.v1#authzed.api.v1.CheckPermissionRequest and 
https://github.com/RedHatInsights/insights-rebac/blob/1af01d1a64cab31d8337c3529febd239594e21fb/api/rebac/v1/relationships.proto#L61

Reply:
Yeah let's be consistent here. We should be able to refer to those same types everywhere, which makes them reusable not just in our own API but in clients. A client should be able to define a SubjectReference or ObjectReference once and reuse it in multiple API calls.

Reply:
Agreed. I will update

Reply:
It should be `relation`. Checks are not limited to relations which model operations; they work for any relation, whether it is one defined via a rule or a tuple.

Reply:
Although one problem with this is that "relation" is arguably defined twice -- it is both the subject relation object together and the kind of relation itself. It's worth a review of what other APIs call this before deciding. I will take a look tomorrow, feel free to have a review too.

Reply:
Quick update-

Zanzibar calls the object/relation/subject an ACL. This is a bit odd IMO.

OpenFGA calls this a relationship. This is a little odd considering relation and relationship are almost synonyms, but it still does work, and it aligns with what you have here. So I say let's go with that. I will update the rest of the terminology I've been using to be consistent and document internally.